src/osd/PGBackend.cc

```diff
@@ -409,7 +409,8 @@ enum scrub_error_type PGBackend::be_compare_scrub_objects(
   if (auth.size != candidate.size) {
     if (error != CLEAN)
       errorstream << ", ";
-    error = SHALLOW_ERROR;
+    if (error != DEEP_ERROR)
+      error = SHALLOW_ERROR;
```

Comment:
Hmm... making shallow take precedence doesn't make sense to me.  Anybody remember why it says that? @athanatos ?